src/seedu/addressbook/data/person/Printable.java

```diff
@@ -0,0 +1,26 @@
+package seedu.addressbook.data.person;
+
+public interface Printable {
+   Name getName();
+   Phone getPhone();
+   Email getEmail();
+   Address getAddress();
+   
+   default String getPrintableString(Name name, Phone phone, Email email, Address address){
+	   StringBuilder stringConcatenation = new StringBuilder();
+	   stringConcatenation.append(name);
+	   
+	   if(!phone.isPrivate()){
+		   stringConcatenation.append(" Phone Number: ").append(phone);
+	   }
+	   
+	   if(!email.isPrivate()){
+		   stringConcatenation.append(" Email Address: ").append(email);
+	   }
+	   
+	   if(!address.isPrivate()){
+		   stringConcatenation.append(" Address: ").append(address);
+	   }
+	   return stringConcatenation.toString();
+   }
```

Comment:
this isn't really what the activity is asking for... `Printable` should be an interface that requires classes that implement it to have the method `getPrintableString` instead of the `getName`, `getPhone` etc. I hope the discussion during tutorial has given you a better idea of how to complete this activity.